scripts/OverlaySurvey.py

```diff
@@ -2,75 +2,105 @@
+  "badResponseNodes": null,
+  "surveyInProgress": true,
+  "topology": {
+    "GBBNXPPGDFDUQYH6RT5VGPDSOWLZEXXFD3ACUPG5YXRHLTATTUKY42CL": null,
+    "GDEXJV6XKKLDUWKTSXOOYVOYWZGVNIKKQ7GVNR5FOV7VV5K4MGJT5US4": {
+      "inboundPeers": [
+        {
+          "averageLatencyMs": 23,
+          "bytesRead": 26392,
+          "bytesWritten": 26960,
+          "duplicateFetchBytesRecv": 0,
+          "duplicateFetchMessageRecv": 0,
+          "duplicateFloodBytesRecv": 10424,
+          "duplicateFloodMessageRecv": 43,
+          "messagesRead": 93,
+          "messagesWritten": 96,
+          "nodeId": "GBBNXPPGDFDUQYH6RT5VGPDSOWLZEXXFD3ACUPG5YXRHLTATTUKY42CL",
+          "secondsConnected": 22,
+          "uniqueFetchBytesRecv": 0,
+          "uniqueFetchMessageRecv": 0,
+          "uniqueFloodBytesRecv": 11200,
+          "uniqueFloodMessageRecv": 46,
+          "version": "v12.2.0-46-g61aadd29"
+        },
+        {
+          "averageLatencyMs": 213,
+          "bytesRead": 32204,
+          "bytesWritten": 31212,
+          "duplicateFetchBytesRecv": 0,
+          "duplicateFetchMessageRecv": 0,
+          "duplicateFloodBytesRecv": 11200,
+          "duplicateFloodMessageRecv": 46,
+          "messagesRead": 115,
+          "messagesWritten": 112,
+          "nodeId": "GBUICIITZTGKL7PUBHUPWD67GDRAIYUA4KCOH2PUIMMZ6JQLNVA7C4JL",
+          "secondsConnected": 23,
+          "uniqueFetchBytesRecv": 176,
+          "uniqueFetchMessageRecv": 2,
+          "uniqueFloodBytesRecv": 14968,
+          "uniqueFloodMessageRecv": 62,
+          "version": "v12.2.0-46-g61aadd29"
+        }
+      ],
+      "numTotalInboundPeers": 2,
+      "numTotalOutboundPeers": 0,
+      "maxInboundPeerCount": 64,
+      "maxOutboundPeerCount": 8,
+      "addedAuthenticatedPeers" : 0,
+      "droppedAuthenticatedPeers" : 0,
+      "p75SCPFirstToSelfLatencyMs" : 121,
+      "p75SCPSelfToOtherLatencyMs" : 112,
+      "lostSyncCount" : 0,
+      "isValidator" : false,
+      "outboundPeers": null
     }
+  }
 }
 """
 
 import argparse
 from collections import defaultdict
 import json
+import logging
 import networkx as nx
+import random
 import requests
 import sys
 import time
 
 import overlay_survey.simulation as sim
+import overlay_survey.util as util
+
+logger = logging.getLogger(__name__)
 
 # A SurveySimulation, if running in simulation mode, or None otherwise.
 SIMULATION = None
 
+# Maximum duration of collecting phase in minutes. This matches stellar-core's
+# internal limit.
+MAX_DURATION = 30
+
+# Maximum number of consecutive rounds in which the surveyor neither sent
+# requests to nor received responses from any nodes. A round contains a batch of
+# requests sent to select nodes, followed by a wait period of 15 seconds,
+# followed by checking for responses and building up the next batch of requests
+# to send. Therefore, a setting of `8` is roughly 2 minutes of inactivity
+# before the script considers the survey complete.
+MAX_INACTIVE_ROUNDS = 8
```

Comment:
Do we need this for cases where we don't get complete survey back from the network? Otherwise, we'd mark the survey complete once we've discovered and surveyed all the peers, right?

Reply:
That's exactly right. I added an explanation to this comment to clarify why `MAX_INACTIVE_ROUNDS` is necessary.